src/components/ToggleSwitch/index.tsx

```diff
@@ -0,0 +1,222 @@
+  circleInActiveColor = '#2C67BC',
+  backgroundActive = '#F5F5F5',
+  backgroundInactive = '#F5F5F5',
+  styleLabelLeft,
+  styleLabelRight,
+  activeTextColor = '#00ADB5',
+  inactiveTextColor = '#2C67BC',
+  containerStyle,
+  labelOff = 'Off',
+  labelOn = 'On'
+}) => {
+  const positionButton = React.useMemo(() => new Animated.Value(0), []);
+
+  const startAnimToOff = () => {
+    Animated.timing(positionButton, {
+      toValue: 0,
+      duration: 150,
+      easing: Easing.ease,
+      useNativeDriver: false
+    }).start();
+  };
+
+  const startAnimToOn = () => {
+    Animated.timing(positionButton, {
+      toValue: 1,
+      duration: 150,
+      easing: Easing.ease,
+      useNativeDriver: false
+    }).start();
+  };
+
+  const positionInterPol = positionButton.interpolate({
+    inputRange: [0, 1],
+    outputRange: [0, dimen.normalizeDimen(22)]
+  });
+  const backgroundColorAnim = positionButton.interpolate({
+    inputRange: [0, 1],
+    outputRange: [backgroundInactive, backgroundActive]
+  });
+
+  const initialOpacityOn = positionButton.interpolate({inputRange: [0, 1], outputRange: [0, 1]});
+  const initialOpacityOff = positionButton.interpolate({inputRange: [0, 1], outputRange: [1, 0]});
+
+  const onPress = () => {
+    if (!value) {
+      startAnimToOff();
+    } else {
+      startAnimToOn();
+    }
+    if (onValueChange) onValueChange(!value);
+  };
+
+  if (value) {
+    startAnimToOn();
+  } else {
+    startAnimToOff();
+  }
+
+  return (
+    <TouchableOpacity
+      style={[styles.container, containerStyle]}
+      activeOpacity={0.9}
+      onPress={onPress}>
+      <Text style={[styles.labelLeft, styleLabelLeft]}>{labelLeft}</Text>
+      <Animated.View
+        style={[
+          styles.mainStyes,
+          {
+            backgroundColor: backgroundColorAnim
+          }
+        ]}>
+        <Animated.Text
+          style={[
+            styles.textOn,
+            {
+              opacity: initialOpacityOn,
+              color: activeTextColor
+            }
+          ]}>
+          {labelOn}
+        </Animated.Text>
+        <Animated.Text
+          style={[
+            styles.textOff,
+            {
+              opacity: initialOpacityOff,
+              color: inactiveTextColor
+            }
+          ]}>
+          {labelOff}
+        </Animated.Text>
+        <Animated.View
+          style={[
+            styles.basicStyle,
+            {
+              transform: [
+                {
+                  translateX: positionInterPol
+                }
+              ],
+              backgroundColor: positionButton.interpolate({
+                inputRange: [0, 1],
+                outputRange: [circleInActiveColor, circleActiveColor]
+              })
+            }
+          ]}
+        />
+      </Animated.View>
+      <Text style={[styles.labelRight, styleLabelRight]}>{labelRight}</Text>
+    </TouchableOpacity>
+  );
+};
+
+export default ToggleSwitch;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new `ToggleSwitch` component looks well-implemented overall. However, there's a potential issue with the way animations are started in this component. The `startAnimToOn` and `startAnimToOff` functions are called directly within the render method (lines 161-165), which means they will be called every time the component re-renders. This could lead to unnecessary animations and performance issues.

Instead of starting the animations directly in the render method, consider using the `useEffect` hook to start the animations only when the `value` prop changes. Here's how you can do it:

```diff
-  if (value) {
-    startAnimToOn();
-  } else {
-    startAnimToOff();
-  }
+  React.useEffect(() => {
+    if (value) {
+      startAnimToOn();
+    } else {
+      startAnimToOff();
+    }
+  }, [value]);
```

This change ensures that the animations are only started when the `value` prop changes, rather than on every render.


<!-- This is an auto-generated comment by OSS CodeRabbit -->